packages/astro/test/fixtures/content-layer/src/content/config.ts

```diff
@@ -18,4 +18,52 @@ const dogs = defineCollection({
 		temperament: z.array(z.string())
 	}),
 })
-export const collections = { blog, dogs };
+
+const cats = defineCollection({
+	type: "experimental_data",
+	loader: async function() {
+		return  [{
+				"breed": "Siamese",
+				"id": "siamese",
+				"size": "Medium",
+				"origin": "Thailand",
+				"lifespan": "15 years",
+				"temperament": ["Active", "Affectionate", "Social", "Playful"]
+			},
+			 {
+				"breed": "Persian",
+				"id": "persian",
+				"size": "Medium",
+				"origin": "Iran",
+				"lifespan": "15 years",
+				"temperament": ["Calm", "Affectionate", "Social"]
+			},
+			 {
+				"breed": "Tabby",
+				"id": "tabby",
+				"size": "Medium",
+				"origin": "Egypt",
+				"lifespan": "15 years",
+				"temperament": ["Curious", "Playful", "Independent"]
+			},
+			{
+				"breed": "Ragdoll",
+				"id": "ragdoll",
+				"size": "Medium",
+				"origin": "United States",
+				"lifespan": "15 years",
+				"temperament": ["Calm", "Affectionate", "Social"]
+			}
+		];
+	},
+	schema: z.object({
```

Comment:
The schema shapes an entry, and the loader returns an array. I wonder why this is the case. Wouldn't it make more sense to match the schema with the data returned by the loader? Why are we forced to return an array and not a single object?

Reply:
The schema is for each item though. The array is for the whole collection

Reply:
Do you mean returning the collection as an object with the ids as keys, or something else?

Reply:
IMHO, if a `schema` defines an object, I would expect the `loader` to return an object. If a `schema` defines an array, I would expect the `loader` to return an array.

However, if we decide to go with the current design, I think we should rename `schema` and `loader`, for example, `entrySchema` or `entitySchema`, and `entitiesLoader` or `entriesLoader`. 

Anyway, not a blocker, we can merge this PR any time

Reply:
This is how collections work now though. The schema is for the individual items, not the whole collection array